### DISCUSSION TITLE
Performance hints for environment effects

### DISCUSSION THREAD

**Discussion by Zireael07** [godotengine/godot-proposals#4384] [Editor]
Tweaking my main project in 4.0 alphas, I discovered that the following effects have a big impact on performance, even at their individual settings turned down:

- SSIL
- SSAO
- SDFGI

(Those three were somewhat expected, I think by now they have a performance cost caveat mentioned in the docs somewhere)

.... and....
- SSR
This was a surprise, and a nasty surprise, I was ready to blame my new rain particles for the sharp drop from 60 to 30 fps...

On reflection (pun intended), I'm guessing this is because they postprocess the entire scene/camera, the bigger the resolution the bigger the drop. To be frank, I'm running Intel Kaby Lake, but still...

Vulkan has a visual profiler, unlike OpenGL 3.0, but it's still fairly limited. It does not call out postprocessing as such, let alone say which one takes X time... still very limited compared to what we get in Unity or UE4

We need performance hints of some sort that tell us that those four (and possibly volumetric fog, although I haven't used it so can't tell) should be the first to go if you want to improve performance. I'm thinking making the enabled boolean tickbox bold and/or colored red or applying the red color to the environment headings themselves... or if that's not possible, (Slow) (Slowest) tags on the tickbox or heading similar to: https://github.com/godotengine/godot/pull/58766 which is for individual enums

PS. Personally I was able to counteract most of the perf drop by using FSR at 0.65 scale (no apparent visual change in my admittedly lowpoly game, but FPS increased again to 45-55)

**Comment by Calinou** [comment]
The [Optimization](https://docs.godotengine.org/en/latest/tutorials/performance/index.html) tutorials are likely the way to go for this kind of stuff. Feel free to open a pull request to improve them :slightly_smiling_face: 

> I'm thinking making the enabled boolean tickbox bold and/or colored red or applying the red color to the environment headings themselves... or if that's not possible, (Slow) (Slowest) tags on the tickbox or heading similar to: https://github.com/godotengine/godot/pull/58766 which is for individual enums

Unfortunately, this can only be done for enums, not boolean values.

> To be frank, I'm running Intel Kaby Lake, but still...

I would highly recommend disabling all the effects you mentioned on integrated graphics. [It has been discussed whether this should be done automatically by default.](https://github.com/godotengine/godot-proposals/issues/2183)
